src/seahawk/seahawk_deck/keyboard.py

```diff
@@ -35,60 +35,82 @@
     """
     Class that actively reads keyboard input
     """
 
     def __init__(self, settings):
         """
-        Initialize 'input_keyboard' node
+        Initialize 'keyboard' node
         """
-        super().__init__("input_keyboard")
+        super().__init__("keyboard")
     
-        # Create publisher to topic 'key_press'
+        # Create publisher to topic 'keystroke'
         self.__key_pub = self.create_publisher(String, "keystroke", 10)
 
+        # Set up client to remotely set parameters on 'pilot_input' node using a service
+        self.__set_params_input_xbox_one = CliRemoteParams(self, "pilot_input")
+
         # Get current user terminal settings and save them for later
         self.__settings = settings
+
+        # Store the current key
+        self.__cur_key = ""
     
     def __get_key(self) -> str:
         """
-        Extracts a single keystroke from the user
-
-        Returns:
-            String of the name of the key which was pressed
+        Extracts a single keystroke from the user and sets private attribute __cur_key
         """
         # Enable raw mode. In raw mode characters are directly read from and written 
         # to the device without any translation or interpretation by the operating system
         tty.setraw(sys.stdin.fileno())
 
         # Read at most one byte (the length of one character) from stdin and decode it
-        key = os.read(sys.stdin.fileno(), 1).decode()
+        self.__cur_key = os.read(sys.stdin.fileno(), 1).decode()
 
         # Make terminal not break
         termios.tcsetattr(sys.stdin, termios.TCSADRAIN, self.__settings)
 
-        return key
-
-    def pub_callback(self):
+    def callback(self):
         """
-        Gets a key from the user publishes it to 'keystroke' topic
+        Gets a key from the user publishes it to 'keystroke' topic and update any parameters
 
         Throws 'KeyboardInterrupt' if user enters ctrl-c
         """
-        key = self.__get_key()
+        self.__get_key()
 
         # If 'ctrl-c' raise 'KeyboardInterrupt'
-        if key == "\x03":
+        if self.__cur_key == "\x03":
             raise KeyboardInterrupt("Terminated the process with ctrl-c")
         
+        self.__pub_callback()
+        self.__param_callback()
+
+    def __pub_callback(self):
+        """
+        Publishes the current key to 'keystroke' topic
+
+        Throws 'KeyboardInterrupt' if user enters ctrl-c
+        """
         # Create and publish message
         msg = String()
-        msg.data = key
+        msg.data = self.__cur_key
         self.__key_pub.publish(msg)
+    
+    def __param_callback(self):
+        """
+        Updates parameters for 'throttle_curve_choice'
+        """
+        # If the user pressed '0', '1', or '2' send that information to the __set_params_input_xbox_one
+        if self.__cur_key in ["0", "1", "2"]:
+            self.__set_params_input_xbox_one.update_params("throttle_curve_choice", self.__cur_key)
```

Comment:
This is a very unergonomic method of keybinding. If we wanted to change these to, say, letters instead, then the letter would be sent. It's better to abstract this I'd say. Consider using a dict, or the index in the array, and sending numbers 0-3 instead of characters "0"-"3", or using id-like names (`"normal"`, `"gentle"`, `"deadzone-50"`...).

Reply:
[wont-fix] This will be updated in the gui implementation of the keyboard, no use fixing something which will be re-written #186